client/data/agency-dashboard/use-fetch-test-connection.ts

```diff
@@ -1,54 +1,61 @@
-import { useQuery, useQueryClient } from '@tanstack/react-query';
-import { useContext } from 'react';
-import SitesOverviewContext from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/context';
+import { useQueries, useQuery } from '@tanstack/react-query';
 import { Site } from 'calypso/jetpack-cloud/sections/agency-dashboard/sites-overview/types';
 import wpcom from 'calypso/lib/wp';
 
-const useFetchTestConnection = (
+// We don't want to trigger another API request to the /test-connection endpoint for a given site
+// one minute after the first successful one.
+const STALE_TIME = 1000 * 60;
+
+const getQueryKey = ( ID: Site[ 'blog_id' ] ) => [ 'jetpack-fetch-test-connection', ID ];
+
+const queryOptions = {
+	refetchOnWindowFocus: false,
+	staleTime: STALE_TIME,
+};
+
+const createQueryFn =
+	( siteId: Site[ 'blog_id' ], isConnectionHealthy: Site[ 'is_connection_healthy' ] ) => () =>
+		wpcom.req.get(
+			{
+				path: `/jetpack-blogs/${ siteId }/test-connection`,
+				apiNamespace: 'rest/v1.1',
+			},
+			{
+				// We call the current health state "stale", as it might be different than the actual state.
+				is_stale_connection_healthy: Boolean( isConnectionHealthy ),
+			}
+		);
+
+export const useFetchTestConnections = ( isPartnerOAuthTokenLoaded: boolean, sites: Site[] ) => {
```

Comment:
I introduced this hook because given the way the dashboard is designed we need both a way to retrieve the connection status for single sites (via `SiteTableRow` or `SiteCard` on mobile) and the list of sites (via `Edit All`, that means bulk select functionality). For the latter we can't use the result of `useFetchTestConnection`. – as both hooks use the same query key they share the query cache.